backend/etcd/etcd.go

```diff
@@ -36,7 +55,50 @@ func (b *Backend) Get(ctx context.Context, key string) ([]byte, error) {
 	return resp.Kvs[0].Value, nil
 }
 
+func (b *Backend) fetchTree(ctx context.Context) error {
+	resp, err := b.client.KV.Get(ctx, b.prefix, clientv3.WithPrefix())
+	if err != nil {
+		return err
+	}
+
+	b.cache = make(map[string][]byte)
+
+	for _, kv := range resp.Kvs {
+		b.cache[strings.TrimLeft(string(kv.Key), b.prefix+"/")] = kv.Value
+	}
+
+	return nil
+}
+
+func (b *Backend) fromCache(ctx context.Context, key string) ([]byte, error) {
+	v, ok := b.cache[key]
+	if !ok {
+		return nil, backend.ErrNotFound
+	}
+
+	return v, nil
+}
+
 // Name returns the name of the backend.
 func (b *Backend) Name() string {
 	return "etcd"
 }
+
+// Option is used to configure the Consul backend.
+type Option func(*Backend)
+
+// WithPrefix is used to specify the prefix to prepend on every keys.
+func WithPrefix(prefix string) Option {
+	return func(b *Backend) {
+		b.prefix = prefix
+	}
+}
+
+// WithPrefetch is used to prefetch the entire tree and cache it to
+// avoid further round trips. If the WithPrefix option is used, will fetch
+// the tree under the specified prefix.
+func WithPrefetch() Option {
+	return func(b *Backend) {
+		b.prefetch = true
+	}
+}
```

Comment:
The code related to the Option type should be moved somewhere else. It seems to be duplicated from the consul package.
WDYT ?

Reply:
:book: [`A little copying is better than a little dependency.`](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=9m28s)

Basically, moving the Option system out of the backends would mean:
- tight coupling between them
- having one backend struct shared between all the backends
- having to export the fields ex: `prefetch` -> `Prefetch`
- having options that only make sense in one backend but not another, but being able to pass them anyway
- and so on

Reply:
I wasn't aware about this video. Thanks ❤️